### PR TITLE
Account linking updates

### PR DESCRIPTION
**Pull request by mpaktiti**
Due to the [deprecation of ID Tokens as authentication credentials for the Mgmt API](https://auth0.com/docs/migrations#deprecating-the-usage-of-id-tokens-on-the-auth0-management-api), the account linking docs need to be updated. 

Some other changes are introduced to the processes as well.

### High-level description of the new processes:

**Admin account linking**
* Requires an admin token (`update:user`) scope
* Can send `user_id` in payload
* Can send `id_token` in payload

**End user account linking**
* Requires an end user token (`update:current_user`) scope
* CANNOT send `user_id` in payload
* Can send `id_token` in payload *BUT*:
  * The `id_token` needs to the signed with RS256
  * The `aud` claim in the `id_token` must identify the client, and be the same client that is the `azp` of the access token used to make the request

### PR changes summary
- [x] Remove article [Link Accounts using Authentication API (Deprecated)](https://auth0.com/docs/link-accounts/auth-api) since the Mandatory Opt-In date for this migration has long passed (`2017-03-01`). Adding a redirect to the main Account Linking doc
- [x] Update **Linking User Accounts** doc with new flows
- [x] Update **Linking User Accounts** doc with section on Auth0.js and how to use it for account linking
- [x] Update **Account Linking Using Server Side Code** doc: remove guidance for embedded Lock and advise for universal login instead, copyedits
- [x] Update **User Initiated Account Linking** doc: update title to **Account Linking Using Client Side Code** for consistency, update snippets with new flow, copyedits
- [x] Migration guide for Account Linking

### Review app
- [Linking User Accounts](https://auth0-docs-content-pr-5930.herokuapp.com/docs/link-accounts)
- [Account Linking Using Client Side Code](https://auth0-docs-content-pr-5930.herokuapp.com/docs/link-accounts/user-initiated-linking)
- [Account Linking Using Server Side Code](https://auth0-docs-content-pr-5930.herokuapp.com/docs/link-accounts/suggested-linking)
- [Migration Guide: Account Linking and ID Tokens](https://auth0-docs-content-pr-5930.herokuapp.com/docs/migrations/guides/account-linking)

### CODE REVIEW

**Comment by mpaktiti** [comment]
@cabralmartin I'm still working on the migration guide but since this has been enabled for new customers by default, I thought we could decouple these two: have you review these updates, while I'm working on the migration guide on another PR.

**Comment by mpaktiti** [comment]
@cabralmartin Migration guide added

File: articles/link-accounts/suggested-linking/index.md [hunk 1]
```diff
@@ -80,7 +51,7 @@ router.get('/suggested-users',ensureLoggedIn, (req,res) => {
 });
 ```
 
-To get a list of all of the user records with the same email address, your app calls the Management API's [Get Users By Email endpoint](/api/v2#!/users-by-email/) using a [Management API Token](/api/management/v2/tokens) with the `read:users` scope:
+To get a list of all of the user records with the same email address, your app calls the Management API's [Get Users By Email endpoint](/api/v2#!/users-by-email/) using a [Management API Access Token](/api/management/v2/tokens) with the `read:users` scope.
```
Comment (cabralmartin) [inline articles/link-accounts/suggested-linking/index.md:58]: 💯

File: articles/migrations/guides/account-linking.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+toc: true
+---
+
+# Migration Guide: Account Linking and ID Tokens
+
+This guide is part of the [Deprecating the usage of ID Tokens on the Auth0 Management API](/migrations#deprecating-the-usage-of-id-tokens-on-the-auth0-management-api) migration, and focuses on the [account linking process](/link-accounts).
+
+For some use cases you could use [ID Tokens](/tokens/id-token) to [link and unlink user accounts](/link-accounts). This functionality is being deprecated. You will have to use [Access Tokens](/tokens/access-token) in all cases.
+
+The grace period for this migration starts on **March 31, 2018** and ends on **March 31, 2019**. From that day forward you will no longer be able to use ID Tokens to link or unlink accounts.
```
Comment (cabralmartin) [inline articles/migrations/guides/account-linking.md]: Dates we can remove until we have the calendar.

File: articles/migrations/guides/account-linking.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
+---
+title: Migration Guide: Account Linking and ID Tokens
+description: Auth0 is deprecating the usage of ID Tokens in the Account Linking process. This article will help you migrate your solution from the old implementation to the new one.
+toc: true
+---
+
+# Migration Guide: Account Linking and ID Tokens
+
+This guide is part of the [Deprecating the usage of ID Tokens on the Auth0 Management API](/migrations#deprecating-the-usage-of-id-tokens-on-the-auth0-management-api) migration, and focuses on the [account linking process](/link-accounts).
```
Comment (cabralmartin) [inline articles/migrations/guides/account-linking.md:9]: Nitpicking, we are deprecating the use of id_tokens in Authorization headers, not in the payload.

File: articles/migrations/guides/account-linking.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+For some use cases you could use [ID Tokens](/tokens/id-token) to [link and unlink user accounts](/link-accounts). This functionality is being deprecated. You will have to use [Access Tokens](/tokens/access-token) in all cases.
+
+The grace period for this migration starts on **March 31, 2018** and ends on **March 31, 2019**. From that day forward you will no longer be able to use ID Tokens to link or unlink accounts.
+
+This article will help you migrate your implementation. First, we will see which use cases are affected. We will continue with reviewing how you can use [scopes](/scopes) to get tokens with different access rights, and how you can use them in the account linking process.
+
+## Summary of changes
```
Comment (cabralmartin) [inline articles/migrations/guides/account-linking.md]: 💯

File: articles/migrations/guides/calling-api-with-idtokens.md [hunk 5]
```diff
@@ -8,9 +8,9 @@ toc: true
 
 For some use cases you could use [ID Tokens](/tokens/id-token) in order to call some of the [/users](/api/management/v2#!/Users/get_users_by_id) and [/device-credentials](/api/management/v2#!/Device_Credentials/get_device_credentials) endpoints of the Management API. This functionality is being deprecated. You will have to use proper [Access Tokens](/tokens/access-token) in order to access any of the endpoints of the [Management API](/api/management/v2).
 
-The grace period for this migration starts on **March 31, 2018** and ends on **March 31, 2019**. That day forward you will no longer be able to use ID Tokens as credentials for the Management API.
+The grace period for this migration starts on **March 31, 2018** and ends on **March 31, 2019**. From that day forward you will no longer be able to use ID Tokens as credentials for the Management API.
```
Comment (cabralmartin) [inline articles/migrations/guides/calling-api-with-idtokens.md]: We can leave dates out.

File: articles/migrations/guides/account-linking.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>You link user accounts with the <a href="/api/management/v2#!/Users/post_identities">Link a user account</a> endpoint of the Management API, and you send the primary account's ID Token in the <code>Authorization</code> header.</td>
+      <td><div class="label label-warning">Affected</div></td>
+      <td><a href="#link-current-user-accounts-with-the-api">Link current user accounts with the API</a></td>
+    </tr>
+    <tr>
+      <td>You link user accounts with the <a href="/api/management/v2#!/Users/post_identities">Link a user account</a> endpoint of the Management API, you send an Access Token in the <code>Authorization</code> header, and the secondary account's <code>user_id</code> in the payload</td>
```
Comment (cabralmartin) [inline articles/migrations/guides/account-linking.md]: Clarify, users won't be affected if the access token contains scope `update:users`. If not, they are affected.

File: articles/migrations/guides/account-linking.md [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+      <td><div class="label label-primary">New use case</div></td>
+      <td><a href="#link-any-user-account-with-the-api">Link any user account with the API</a></td>
+    </tr>
+    <tr>
+      <td>You link user accounts with the <a href="/auth0js#user-management">Auth0.js library</a>, and you use the primary account's ID Token to instantiate <code>auth0.Management</code></td>
+      <td><div class="label label-warning">Affected</div></td>
+      <td><a href="#link-current-user-accounts-with-auth0-js">Link current user accounts with Auth0.js</a></td>
+    </tr>
+    <tr>
+      <td>You link user accounts with the <a href="/auth0js#user-management">Auth0.js library</a>, and you use an Access Token to instantiate <code>auth0.Management</code></td>
```
Comment (cabralmartin) [inline articles/migrations/guides/account-linking.md]: Same as above.
access token + update:users + user_id in payload = OK
access token + update:current_user_* : user_id in payload = NOT OK

**Review by cabralmartin — CHANGES_REQUESTED** [review]

File: articles/migrations/guides/account-linking.md (hunk 6 above)
Comment (mpaktiti) [reply articles/migrations/guides/account-linking.md]: @cabralmartin If I get this right, if you have an access token with `update:current_user_* ` scope
- it is not ok to send the user_id in payload
- it is ok to send the SECONDARY_ACCOUNT_ID_TOKEN in the payload
?

Comment (cabralmartin) [reply articles/migrations/guides/account-linking.md]: Correct!

**Pull request merged**